### DISCUSSION TITLE
Behavior of the "skip" for multiple columns in "or" operator in where clause

### DISCUSSION THREAD

**Discussion by alisabrigok** [ujjwalguptaofficial/JsStore#300] [Q&A] [answered]
When I query having multiple columns in "or" operator in the where clause, and if I use "skip", then "skip" gets applied to results of the all columns.

Explaining by example using IDBStudio with demo database:

Below returns 14 results; 
- 2 rows belong to customerId 81
- 1 row belongs to orderId 10256
- 11 rows belong to employeeId 5

```
{
    select({
        from: "Orders",
        where: {
            employeeId: 5,
            or: {
                customerId: 81,
                orderId: 10256,
            },
        },
    })
}
```

If I apply "skip: 1" to this exact query as above, then it returns 11 results. 
But I was expecting it to return 13 results because I just wanted to skip 1 row. Is my expectation correct here?

Here I see that each of the columns loses a row due to the applied skip;
- 1 row left from customerId 81
- No row left from orderId 10256
- 10 rows left from employeeId 5

```
{
    select({
        from: "Orders",
        where: {
            employeeId: 5,
            or: {
                customerId: 81,
                orderId: 10256,
            },
        },
        skip: 1
    })
}
```

So skip here is applied to the records related to all of the columns.
Do you think this Is an expected behavior or is this a bug? @ujjwalguptaofficial 

Thanks for all the help.

**Comment by ujjwalguptaofficial** [comment]
this is a bug, i have created an issue for it - https://github.com/ujjwalguptaofficial/JsStore/issues/301

**Reply by ujjwalguptaofficial** [reply]
@alisabrigok fixed the issue in v - 4.4.6. Please check and let me know.

**Reply by alisabrigok** [reply]
I guess you haven't released a new version to IDBStudio, the issue still exists there.

But I tried 4.4.6 in my local environment, and right now there's a slightly different issue happening:

If I have multiple "or" just like I mentioned in the description and let's say this query has actually 45 results:

- I am limiting by 20 and skipping by 0 - I get 20 results,
- Then while still limiting by 20, if I skip by 20 - I get the exact same 20 results as the first one
- If I skip by 40 - I again get the exact same 20 results (I should have gotten 5)

So skip doesn't seem to work now. This time it just gives me the same result every time.

Can you please try this scenario? @ujjwalguptaofficial 

If you release a new version to IDBStudio, I can also check from there and give you a reproducible example.

Thanks.

**Reply by ujjwalguptaofficial** [reply]
@alisabrigok updated idbstudio, please reproduce the error and let me know.

**Comment by alisabrigok** [comment]
Here's a reproducible example:

Below returns 83 records:

```
{
    select({
        from: "Orders",
        where: [
                {
                    shipperId: 2,
                    or: {
                        employeeId: 5,
                        customerId: 88,
                    },
                }
            ],
    })
}
```

If I add skip 0 and limit 20 to the same query, I get the first 20 results, which is correct:

```
{
    select({
        from: "Orders",
        where: [
                {
                    shipperId: 2,
                    or: {
                        employeeId: 5,
                        customerId: 88,
                    },
                }
            ],
        limit: 20,
        skip: 0
    })
}
```

But if I try to skip the first 20 and get the next 20 records, it doesn't work. I still get the first 20 results no matter what the skip count is. 

Below returns the exact same records as above:

```
{
    select({
        from: "Orders",
        where: [
                {
                    shipperId: 2,
                    or: {
                        employeeId: 5,
                        customerId: 88,
                    },
                }
            ],
        limit: 20,
        skip: 20
    })
}
```

It seems that skip is ignored when where clause is used in array form.

I hope this example helps you @ujjwalguptaofficial 

Thanks a lot.

**Reply by ujjwalguptaofficial** [reply]
thanks fix is on the way.

**Reply by ujjwalguptaofficial** [reply]
@alisabrigok fixed in v - 4.4.7. Please install and let me know.

**Reply by alisabrigok** [reply]
@ujjwalguptaofficial Awesome! 

In my local environment, everything seems to be working as it should. 
Thanks for the swift fixes for the past 2 days. Great job! :clap: :clap:

**Reply by ujjwalguptaofficial** [reply]
Thanks @alisabrigok for the great explanation of issues, i was able to fix this quickly because of the well explanation and easy reproducability. :clap:

**Reply by ujjwalguptaofficial** [reply]
i have also updated idbstudio with new version of jsstore.

**Reply by alisabrigok** [reply]
I'm glad to hear my explanations helped you. I did my best to make it easier for you to debug so that you can spend your minimum amount of time on these. Being an open-source maintainer already isn't easy, it's the least we can do to help.

Btw, tested on IDBStudio now, and all looks good :thumbsup:

**Comment by alisabrigok** [comment, marked as answer]
The issue is fixed in version 4.4.7, kudos to @ujjwalguptaofficial
